lib/stm32/common/st_usbfs_core.c

```diff
@@ -37,44 +37,56 @@ void st_usbfs_set_address(usbd_device *dev, uint8_t addr)
 	SET_REG(USB_DADDR_REG, (addr & USB_DADDR_ADDR) | USB_DADDR_EF);
 }
 
-/**
- * Set the receive buffer size for a given USB endpoint.
- *
- * @param dev the usb device handle returned from @ref usbd_init
- * @param ep Index of endpoint to configure.
- * @param size Size in bytes of the RX buffer. Legal sizes : {2,4,6...62}; {64,96,128...992}.
- * @returns (uint16) Actual size set
+/*
+ * Convert from raw byte count to special encoded value suitable for writing
+ * to USB_COUNTn_RX registers, which consists of NUM_BLOCK[4:0] and BLSIZE.
+ * Returns the corresponding actual size in *real_size.
  */
-uint16_t st_usbfs_set_ep_rx_bufsize(usbd_device *dev, uint8_t ep, uint32_t size)
+static uint32_t convert_bufsize(uint32_t size, uint16_t *real_size)
 {
-	uint16_t realsize;
-	(void)dev;
+	uint32_t encoded;
 	/*
 	 * Writes USB_COUNTn_RX reg fields : bits <14:10> are NUM_BLOCK; bit 15 is BL_SIZE
 	 * - When (size <= 62), BL_SIZE is set to 0 and NUM_BLOCK set to (size / 2).
 	 * - When (size > 62), BL_SIZE is set to 1 and NUM_BLOCK=((size / 32) - 1).
 	 *
 	 * This algo rounds to the next largest legal buffer size, except 0. Examples:
-	 *	size =>	BL_SIZE, NUM_BLOCK	=> Actual bufsize
-	 *	0		0		0			??? "Not allowed" according to RM0091, RM0008
-	 *	1		0		1			2
-	 *	61		0		31			62
-	 *	63		1		1			64
+	 *	size =>		BL_SIZE,	NUM_BLOCK	=> Actual bufsize
+	 *	0		0		0		??? "Not allowed" according to RM0091, RM0008
+	 *	1		0		1		2
+	 *	61		0		31		62
+	 *	63		1		1		64
 	 */
 	if (size > 62) {
 		/* Round up, div by 32 and sub 1 == (size + 31)/32 - 1 == (size-1)/32)*/
-		size = ((size - 1) >> 5) & 0x1F;
-		realsize = (size + 1) << 5;
+		encoded = ((size - 1) >> 5) & 0x1F;
+		*real_size = (encoded + 1) << 5;
 		/* Set BL_SIZE bit (no macro for this) */
-		size |= (1<<5);
+		encoded |= (1<<5);
 	} else {
 		/* round up and div by 2 */
-		size = (size + 1) >> 1;
-		realsize = size << 1;
+		encoded = (size + 1) >> 1;
+		*real_size = encoded << 1;
```

Comment:
please drop all the whitespace changes.  All you're doing here is adding the "output" parameter right?  Given that the "output" is just the existing return value <<1 I'm kinda curious what the value is at all?

Reply:
Hi, thanks for reviewing. What whitespace changes? I changed the variable name "size" to "encoded" for readability (and not modify the input variable).

The "output" is just the encoded (returned) << 1 in the "else" clause but not in the "then" clause. The "output" may differ from the input "size" since it can be rounded up to next even number.

Reply:
If you like I can do the table layout in a separate commit.